pktoken/pktoken.go

```diff
@@ -188,3 +188,21 @@ func (p *PKToken) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+func (p *PKToken) GetClaimMap() (map[string]interface{}, error) {
+	var claims map[string]interface{}
+	if err := json.Unmarshal(p.Payload, &claims); err != nil {
+		return nil, err
+	}
+	return claims, nil
+}
+
+func (p *PKToken) GetClaim(key string) ([]byte, error) {
+	if payMap, err := p.GetClaimMap(); err != nil {
+		return nil, err
+	} else if value, ok := payMap[key].(string); ok {
+		return []byte(value), nil
+	} else {
+		return nil, fmt.Errorf("claim=(%s) does not exist in ID Token payload", key)
+	}
+}
```

Comment:
I'm fine with not having this in the PK Token, but it is extremely convenient to be able to get claims from the PK Token. Otherwise you just get lines and lines of json.Unmarshal everywhere

What about something like:
```
func (p *PKToken) GetClaim[T any](key string) (T, error) {
  var claims struct {
    Value T key
  }
  if err := json.Unmarshal(pkt.Payload, &claims); err != nil {
    return nil, err
  }
  return claims.Value
}
```

Reply:
I don't think we need it right now and we can revisit it later